### PR TITLE
Remove redundant calls of queueRepositoryUpdateAndModelRefresh (closes #1465)

### LINKED ISSUE #1465: Does `queueRepositoryUpdateAndModelRefresh` need to be called from so many places?
Can't we generally just rely on listening to changes to .git/machete and git repository? 🤔

### PR DESCRIPTION
**Pull request by mkondratek**
Tested so far:
- [x] discover
- [x] slide in (from base slide in)
- [x] slide in (from slide in unmanaged root) 
- [x] slide in (from slide in unmanaged non-root)
- [x] slide out 
- [x] traverse - but not very meticulously

### CODE REVIEW

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/DiscoverAction.java [hunk 1]
```diff
@@ -151,7 +151,6 @@ public void run(ProgressIndicator indicator) {
       @Override
       @UIEffect
       public void onSuccess() {
-        baseEnhancedGraphTable.queueRepositoryUpdateAndModelRefresh();
         VfsUtil.markDirtyAndRefresh(/* async */ false, /* recursive */ true, /* reloadChildren */ false,
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/DiscoverAction.java:154]: this one will trigger an update from the machete file changes listener

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/OpenMacheteFileAction.java [hunk 2]
```diff
@@ -56,6 +56,7 @@ public void actionPerformed(AnActionEvent anActionEvent) {
             /* message */ getString("action.GitMachete.OpenMacheteFileAction.notification.title.cannot-open"))))
         .getOrNull();
 
+    // t0d0: BUG - call from the find action dialog when the file does not exist
     getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/OpenMacheteFileAction.java:60]: Will be addressed on a separate PR

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java [hunk 3]
```diff
@@ -303,6 +303,7 @@ private void trackCurrentBranchChange(GitRepository repository) {
         mostRecentlyCheckedOutBranch = repositoryCurrentBranchName;
       }
     }
+    // required to indicate the currently checked out branch after a checkout
     queueRepositoryUpdateAndModelRefresh();
```
Comment (mkondratek) [inline frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:307]: the whole method run on vcs changes! this one remains for sure!

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/OverrideForkPointBackgroundable.java [hunk 4]
```diff
@@ -55,6 +55,7 @@ private void overrideForkPoint(IManagedBranchSnapshot branch, ICommitOfManagedBr
       setOverrideForkPointConfigValues(myProject, root, branch.getName(), forkPoint, branch.getPointedCommit());
     }
 
+    // required since the change of .git/config is not considered as a change to VCS (and detected by the listener)
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/OverrideForkPointBackgroundable.java:58]: let's put some comments on the calls - probably it could be enforced by some script (?)

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java [hunk 5]
```diff
@@ -450,7 +451,7 @@ private Consumer<IGitMacheteRepositorySnapshot> getSuccessfulDiscoverRepositoryS
       @UI Runnable doOnUIThreadWhenReady) {
     return (IGitMacheteRepositorySnapshot repositorySnapshot) -> ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> {
       gitMacheteRepositorySnapshot = repositorySnapshot;
-      queueRepositoryUpdateAndModelRefresh(doOnUIThreadWhenReady);
+      doOnUIThreadWhenReady.run();
```
Comment (mkondratek) [inline frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:454]: tested - no negative consequences noted

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/OpenMacheteFileAction.java (hunk 2 above)
Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/OpenMacheteFileAction.java]: https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1482

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java [hunk 6]
```diff
@@ -98,27 +98,27 @@ public void execute() {
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java:98]: I think we still need `queueRepositoryUpdateAndModelRefresh` in the case of `traverseNextEntry` since it must depend on the newest repo snapshot.

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/SlideInUnmanagedBelowAction.java [hunk 7]
```diff
@@ -68,25 +68,15 @@ public void actionPerformed(AnActionEvent anActionEvent) {
           branchLayoutWriter,
           preSlideInRunnable,
           slideInOptions,
-          parentName) {
-        @Override
-        public void onFinished() {
-          getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/SlideInUnmanagedBelowAction.java:74]: :benissimo:

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/SlideOut.java [hunk 8]
```diff
@@ -120,15 +120,14 @@ private void suggestBranchDeletion(String branchName, @UI Runnable doInUIThreadW
   private void handleBranchDeletionDecision(String branchName, boolean shouldDelete, @UI Runnable doInUIThreadWhenReady) {
     slideOutBranch(branchName);
     if (shouldDelete) {
-      graphTable.queueRepositoryUpdateAndModelRefresh(
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/SlideOut.java:123]: Is `graphTable` still used?

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/DiscoverAction.java [hunk 9]
```diff
@@ -151,7 +151,6 @@ public void run(ProgressIndicator indicator) {
       @Override
       @UIEffect
       public void onSuccess() {
-        baseEnhancedGraphTable.queueRepositoryUpdateAndModelRefresh();
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/DiscoverAction.java:154]: `BaseEnhancedGraphTable` params are typically just called `graphTable`, rename while you're here

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/DiscoverAction.java:154]: Also... change `runWriteActionOnUIThread` to `blockRunWriteActionOnUIThread` above... now the entire construct doesn't seem to have much sense

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/DiscoverAction.java:154]: Yeah, generally backgroundable in the current form seems redundant TBH, to be fixed within #1465

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToParent.java [hunk 10]
```diff
@@ -85,7 +85,7 @@ public void execute() {
             syncToRemoteRunnable);
         // Note that checking out the branch to be slid out has the unfortunate side effect
         // that we won't suggest deleting the branch after the slide out.
-        checkoutAndExecuteOnUIThread(gitRepository, graphTable, branch.getName(), slideOut);
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToParent.java:88]: Ditto, can `graphTable` be removed from this class/`TraverseSyncToRemote`?

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToParent.java:88]: Oh okay, I see 👇🏻
